### PR TITLE
Ignored case changes in bigquery table schema mode/type

### PR DESCRIPTION
**Pull request by melinath**
BigQuery table schema normalizes the mode and type to uppercase, which causes a permadiff for the end user if they set it in lowercase.

Also removed tests for setting type to nil, because a field type being nil is invalid.

Resolved https://github.com/hashicorp/terraform-provider-google/issues/9472

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
bigquery: Fixed permadiff due to lowercase mode/type in `google_bigquery_table.schema`
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_bigquery_table.go [hunk 1]
```diff
@@ -167,28 +168,32 @@ func bigQueryTableSchemaDiffSuppress(name, old, new string, _ *schema.ResourceDa
...
+		return "NULLABLE"
+	}
+	// Upper-case to get case-insensitive comparisons. https://github.com/hashicorp/terraform-provider-google/issues/9472
+	return strings.ToUpper(mode.(string))
 }
 
-func bigQueryTableModeIsForceNew(old, new interface{}) bool {
-	eq := bigQueryTableModeEq(old, new)
+func bigQueryTableModeIsForceNew(old, new string) bool {
+	eq := old == new
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_bigquery_table.go:196]: This is a safe change because we presanitize the value now, right?

File: mmv1/third_party/terraform/tests/resource_bigquery_table_test.go [hunk 2]
```diff
@@ -34,28 +34,28 @@ func TestBigQueryTableSchemaDiffSuppress(t *testing.T) {
 			ExpectDiffSuppress: false,
 		},
 		"no change": {
-			Old:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"finalKey\" : {} }]",
-			New:                "[{\"name\": \"someValue\", \"anotherKey\" : \"anotherValue\", \"finalKey\" : {} }]",
+			Old:                "[{\"name\": \"someValue\", \"type\": \"INT64\", \"anotherKey\" : \"anotherValue\", \"finalKey\" : {} }]",
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_bigquery_table_test.go:37]: This change is to reflect the actual restrictions in the API and not new validation introduced in this PR, right? That's what I understood "Also removed tests for setting type to nil, because a field type being nil is invalid." to mean, at least

File: mmv1/third_party/terraform/resources/resource_bigquery_table.go (hunk 1 above)
Comment (melinath) [reply mmv1/third_party/terraform/resources/resource_bigquery_table.go:196]: yeah, since we know that they're both strings (and we've already done string.ToUpper) this is safe.

File: mmv1/third_party/terraform/tests/resource_bigquery_table_test.go (hunk 2 above)
Comment (melinath) [reply mmv1/third_party/terraform/tests/resource_bigquery_table_test.go:37]: yeah, the type field is actually required in the API, but a lot of these unit test cases were either missing it or explicitly testing what happened when you deleted it. (But in reality, the API doesn't allow you to leave out the type field.) If we make sure it's present, we don't have to check for nil values.

Comment (melinath) [reply mmv1/third_party/terraform/tests/resource_bigquery_table_test.go:37]: hmmm... I might need to double-check whether setting an invalid schema would cause a panic, actually.

**Review by rileykarson — APPROVED** [review]

File: mmv1/third_party/terraform/tests/resource_bigquery_table_test.go [hunk 3]
```diff
@@ -1050,10 +1098,10 @@ var testUnitBigQueryDataTableIsChangableTestCases = []testUnitBigQueryDataTableJ
 		changeable: false,
 	},
 	{
-		name:       "typeModeOmission",
+		name:       "modeToDefaultNullable",
 		jsonOld:    "[{\"name\": \"someValue\", \"type\" : \"BOOLEAN\", \"mode\" : \"REQUIRED\", \"description\" : \"someVal\" }]",
 		jsonNew:    "[{\"name\": \"someValue\", \"type\" : \"BOOLEAN\", \"description\" : \"some new value\" }]",
-		changeable: false,
+		changeable: true,
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_bigquery_table_test.go:1104]: If mode is omitted, it will be NULLABLE, so I believe this should match typeModeReqToNull (a few cases up) and be changeable. Previously it seems this was not working correctly?

**Comment by melinath** [comment]
@rileykarson I believe the test failures are now unrelated to this PR. I ended up having to make more significant changes than I expected, so I'm re-requesting review.

**Review by rileykarson — APPROVED** [review]
